packages/accounts-controller/src/AccountsController.ts

```diff
@@ -902,6 +904,10 @@ export class AccountsController extends BaseController<
                 importTime: Date.now(),
                 lastSelected,
               },
+              options: {
+                ...account.options,
+                entropySource: added.entropySource,
+              },
```

Comment:
If we go with my other suggestion, this is where we would check if `added.keyring.type === KeyringTypes.Hd` and inject the `entropySource` only for those!

Reply:
done